src/data-model-providers/codegen/ServerClusterInterfaceRegistry.h

```diff
@@ -33,9 +33,10 @@ namespace app {
 struct ServerClusterRegistration
 {
     // A single-linked list of clusters registered for the given `endpointId`
-    ServerClusterInterface * const serverClusterInterface;
+    ServerClusterInterface * serverClusterInterface;
```

Comment:
So we had made this const because mutating this after registration is going to break the world.  It would really be _much_ better to keep this const and construct the registration objects with the right clusters.

For example, instead of having two parallel arrays of OtaProviderServer and ServerClusterRegistration, you could have one array of a struct that:

1. Has those two things as members (or inherits from both, either way; no strong feelings on inheritance vs composition here), with the OtaProviderServer first so it gets constructed first.
2. Has a constructor that constructs the registration by passing in the server instance.

That would allow us to keep more safety/sanity here.

Reply:
I could not find an obvious way to construct-initialize things here. What I would like is to have an array of pairs of "interface + registration" since this is what ember wants: ready-to use registrations of max-endpoint-count.

I will take a look of what options we have, however array initialization (without trying for generated code) is unclear to me ... may need more template magic.